src/protocols/bdx/tests/TestTransferDiagnosticLog.cpp

```diff
@@ -0,0 +1,306 @@
+#include <string.h>
+
+#include <pw_unit_test/framework.h>
+
+#include <protocols/bdx/BdxTransferDiagnosticLog.h>
+#include <system/RAIIMockClock.h>
+#include <system/SystemClock.h>
+#include <system/SystemTimer.h>
+
+using namespace ::chip;
+using namespace ::chip::bdx;
+using namespace ::chip::Protocols;
+using namespace ::chip::System::Clock::Literals;
+
+namespace {
+
+class TestTransferDiagnosticLog : public ::testing::Test
+{
+public:
+    static void SetUpTestSuite() { ASSERT_EQ(Platform::MemoryInit(), CHIP_NO_ERROR); }
+
+    static void TearDownTestSuite() { Platform::MemoryShutdown(); }
+
+protected:
+    TransferSession mTransferSession{};
+};
+
+TEST_F(TestTransferDiagnosticLog, InitsDiagnosticLog)
+{
+    System::Clock::Internal::RAIIMockClock mockClock;
+
+    mTransferSession.WaitForTransfer(TransferRole::kReceiver, TransferControlFlags::kSenderDrive, uint16_t{ 512 }, 1000_ms);
```

Comment:
if we give a name to 512 like `constexpr uint16_t kMaxBlockSize = 512` we both make it clear what it means and do not need a cast.

Do other magic numbers maybe need constant names?